CI builds failing in `tj-actions/changed-files`

In the past couple of days, PR's are failing due to errors that seem to be caused by `tj-actions/changed-files`.
#7132 introduced the use of that action.

Here is a typical failure, from https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390, with the relevant log section extracted:
```
Run tj-actions/changed-files@v34
  with:
    json: true
    separator:  
    include_all_old_new_renamed_files: false
    old_new_separator: ,
    old_new_files_separator:  
    files_separator: 
  
    files_ignore_separator: 
  
    path: .
    quotepath: true
    dir_names: false
    fetch_depth: 40
    max_fetch_depth: 5000
    since_last_remote_commit: false
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.11.0/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.11.0/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.0/x64
    Python[2](https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390#step:29:2)_ROOT_DIR: /opt/hostedtoolcache/Python/[3](https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390#step:29:3).11.0/x6[4](https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390#step:29:4)
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.0/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.11.0/x64/lib
    CP_VERSION: 8.0.0-beta.[5](https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390#step:29:5)-5-ge88f8d[6](https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390#step:29:6)5[7](https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390#step:29:7)
Run # "Calculating the previous and current SHA..."
  # "Calculating the previous and current SHA..."
  bash $GITHUB_ACTION_PATH/diff-sha.sh
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.11.0/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.11.0/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.0/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.0/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.11.0/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.11.0/x64/lib
    CP_VERSION: [8](https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390#step:29:8).0.0-beta.5-5-ge88f8d657
    GITHUB_SERVER_URL: https://github.com
    GITHUB_REPOSITORY: adafruit/circuitpython
    GITHUB_REF: refs/pull/7327/merge
    GITHUB_BASE_REF: main
    GITHUB_HEAD_REF: in-group-remove
    GITHUB_WORKSPACE: /home/runner/work/circuitpython/circuitpython
    GITHUB_EVENT_HEAD_REPO_FORK: true
    GITHUB_EVENT_PULL_REQUEST_BASE_SHA: [9](https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390#step:29:9)06cd6eb65726a9ca04974582344d772a04929df
    GITHUB_EVENT_PULL_REQUEST_HEAD_SHA: 64[12](https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390#step:29:12)d971facb6c0c75aaced897f39dc0a7d[17](https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390#step:29:17)bb2
    GITHUB_EVENT_BEFORE: bd91959ce3eb68375739680bd795fe3f7d0973f0
    GITHUB_EVENT_FORCED: 
    GITHUB_REFNAME: 7327/merge
    INPUT_SHA: 
    INPUT_BASE_SHA: 
    INPUT_SINCE: 
    INPUT_UNTIL: 
    INPUT_PATH: .
    INPUT_FETCH_DEPTH: 40
    INPUT_MAX_FETCH_DEPTH: 5000
    INPUT_SINCE_LAST_REMOTE_COMMIT: false
changed-files-diff-sha
  Verifying git version...
  Valid git version found: (2.38.1)
  Running on a pull request event...
  Fetching remote refs...
  remote: Total 0 (delta 0), reused 0 (delta 0), pack-reused 0        
  From https://github.com/adafruit/circuitpython
   * [new branch]          main       -> origin/main
  branch 'main' set up to track 'origin/main'.
  Fetching 40 commits...
  fatal: error in object: unshallow 98002cf0abe5019cf3c57e6c[18](https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390#step:29:18)96c4c5dd[22](https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390#step:29:22)6f[24](https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390#step:29:24)
  Error: Process completed with exit code 1[28](https://github.com/adafruit/circuitpython/actions/runs/3666904634/jobs/6198998390#step:29:29).
```
Note the `INPUT_FETCH_DEPTH: 40` setting, which is suspiciously the same as the `Fetching 40 commits..` just before the fatal error.

I don't really understand the reason for the error, but I wonder if it is because we are doing shallow clones of `circuitpython`, with `depth=1`. Or perhaps we just need to tweak the `tj-actions/changed-files` parameters.

_*If*_ it is the case that the `depth=1` clones are the issue, here is some more discussion:

Shallow clones are missing information. In #7225, I suggested we might use partial clones instead. They do lazy fetching of a tree, but from a user's point of view they are "complete", fetching more info dynamically as necessary.

There are "treeless" partial clones and "blobless" partial clones. Both are a little slower (2 seconds vs 5 seconds to fetch `circuitpython` for me), but that's not too bad. It's not clear to me which is better for a "fetch, build, and discard" CI job, but there is some discussion here:

https://github.blog/2020-12-21-get-up-to-speed-with-partial-clone-and-shallow-clone/  (detailed overview)
https://gist.github.com/stormwild/b7ae53f401ecf9f51f097a43d01be23b (seems to recommend blobless for CI builds)
I'm sure there are more things to read too.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -33,7 +33,7 @@ jobs:
     - uses: actions/checkout@v3
       with:
         submodules: false
-        fetch-depth: 1
+        fetch-depth: 0
     - name: Set up Python 3
       uses: actions/setup-python@v4
       with:

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -138,7 +138,7 @@ jobs:
     - name: Get changes
       id: get-changes
       if: github.event_name == 'pull_request'
-      uses: tj-actions/changed-files@v34
+      uses: tj-actions/changed-files@v34.5.3
       with:
         json: true
         sha: ${{ steps.get-last-commit-with-checks.outputs.commit && github.event.after }}

